source/base.c

```diff
@@ -57,21 +57,29 @@ bool C2D_Init(size_t maxObjects)
 	C3D_ProcTexCombiner(&ctx->ptBlend, true, GPU_PT_U, GPU_PT_V);
 	C3D_ProcTexFilter(&ctx->ptBlend, GPU_PT_LINEAR);
 
+	C3D_ProcTexInit(&ctx->ptCircle, 0, 1);
+	C3D_ProcTexClamp(&ctx->ptCircle, GPU_PT_MIRRORED_REPEAT, GPU_PT_MIRRORED_REPEAT);
+	C3D_ProcTexCombiner(&ctx->ptCircle, true, GPU_PT_SQRT2, GPU_PT_SQRT2);
+	C3D_ProcTexFilter(&ctx->ptCircle, GPU_PT_LINEAR);
+
 	// Prepare proctex lut
 	float data[129];
 	int i;
 	for (i = 0; i <= 128; i ++)
 		data[i] = i/128.0f;
 	ProcTexLut_FromArray(&ctx->ptBlendLut, data);
 
+	for (i = 0; i <= 128; i ++)
+		data[i] = (i >= 127) ? 0 : 1;
+	ProcTexLut_FromArray(&ctx->ptCircleLut, data);
+
 	ctx->flags = C2DiF_Active;
 	ctx->vtxBufPos = 0;
 	ctx->vtxBufLastPos = 0;
 	Mtx_Identity(&ctx->projMtx);
 	Mtx_Identity(&ctx->mdlvMtx);
 	ctx->fadeClr = 0;
 
-	C3D_FrameEndHook(C2Di_FrameEndHook, NULL);
```

Comment:
Why was this moved?

Reply:
FrameEndHook doesn't get overwritten by anything else, unless you overwrite it yourself.